src/alacris_lib/operator_contract.ml

```diff
@@ -23,6 +24,21 @@ let topic_of_hash (hash : Digest.t) : Bytes.t option =
   Some (encode_function_parameters [abi_digest hash])
 
 
+let retrieve_contract_address_quadruple : Digest.t -> (Address.t * Digest.t * Digest.t * Revision.t) Lwt_exn.t =
```

Comment:
Considering there are two `Digest.t`, maybe make it a structure with named fields, say called `contract_deployment` so there is no confusing the `code_hash` with a `transaction_hash`.

There might be a way to derive equality for it, if it's not automatically done already.

Reply:
I agree that it can be done better.
But I think that it is adequate as it is as it nails the problem on its head.